src/daft-catalog/src/identifier.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::Display;
+
+use crate::errors::{Error, Result};
+
+/// An object's namespace (location).
+pub type Namespace = Vec<String>;
+
+/// An object's name.
+pub type Name = String;
+
+/// A reference (path) to some catalog object.
+#[derive(Debug, Clone)]
+pub struct Identifier {
+    namespace: Namespace,
+    name: Name,
+}
+
+impl Identifier {
+    /// Creates an identifier from its namespace and name.
+    pub fn new(namespace: Namespace, name: Name) -> Self {
+        Self { namespace, name }
+    }
+
+    /// Returns the namespace
+    pub fn namespace(&self) -> &Namespace {
+        &self.namespace
+    }
+
+    /// Returns the name
+    pub fn name(&self) -> &Name {
+        &self.name
+    }
+
+    /// Returns true if this is a qualified identifier e.g. has a namespace.
+    pub fn has_namespace(&self) -> bool {
+        !self.namespace().is_empty()
+    }
+
+    /// Parses an identifier using sqlparser to validate the input.
+    pub fn parse(input: &str) -> Result<Identifier> {
+        // TODO daft should define its own identifier domain.
+        use sqlparser::{dialect::PostgreSqlDialect, parser::Parser};
+        let err = Error::InvalidIdentifier {
+            input: input.to_string(),
+        };
+        let mut parser = match Parser::new(&PostgreSqlDialect {}).try_with_sql(input) {
+            Ok(res) => res,
+            Err(_) => return Err(err),
+        };
+        let parts = match parser.parse_identifiers() {
+            Ok(res) => res,
+            Err(_) => return Err(err),
+        };
```

Comment:
nit: we tend to use `let Ok(...)` or `let Some(...)` when matching on results or options. 
```suggestion
        let Ok(mut parser) = Parser::new(&PostgreSqlDialect {}).try_with_sql(input)  else {

            return Err(err);
        };
        let Ok(parts) = parser.parse_identifiers()  else {
          return Err(err);
        };
```